src/power_process.c

```diff
@@ -667,4 +668,116 @@ void remove_explored_flags_for_power_sight(struct PlayerInfo *player)
+    if (!thing_is_invalid(trgtng))
+    {
+        if ( (creatng->mappos.x.stl.num == trgtng->mappos.x.stl.num) && (creatng->mappos.y.stl.num == trgtng->mappos.y.stl.num) )
+        {
+            if (abs(creatng->mappos.z.val - trgtng->mappos.z.val) <= creatng->solid_size_yz)
+            {
+                timebomb_explode(creatng);
+                return;
+            }
+        }
+    }
+    if (cctrl->timebomb_countdown != 0)
+    {
+        cctrl->timebomb_countdown--;
+    }
+    else
+    {
+        timebomb_explode(creatng);
+    }
+}
+
+void timebomb_explode(struct Thing *creatng)
+{
+    struct SpellConfig* spconf = get_spell_config(SplK_TimeBomb);
+    struct ShotConfigStats* shotst = get_shot_model_stats(spconf->shot_model);
+    SYNCDBG(8, "Explode Timebomb")
+    //struct Thing* castng = creatng; //todo cleanup
+    long weight = compute_creature_weight(creatng);
+    long weight_multiplier = weight / 64;
+    
+    if (shotst->area_range != 0) {
+        struct CreatureStats* crstat = creature_stats_get_from_thing(creatng);
+        struct CreatureControl* cctrl = creature_control_get_from_thing(creatng);
+        long dist = (compute_creature_attack_range(shotst->area_range * COORD_PER_STL, crstat->luck, cctrl->explevel) * weight_multiplier);
+        long damage = (compute_creature_attack_spell_damage(shotst->area_damage, crstat->luck, cctrl->explevel, creatng) * weight_multiplier);
+        HitTargetFlags hit_targets = hit_type_to_hit_targets(shotst->area_hit_type);
+        explosion_affecting_area(creatng, &creatng->mappos, dist, damage, shotst->area_blow * weight_multiplier, hit_targets, shotst->damage_type);
+    }
+
+    create_used_effect_or_element(&creatng->mappos, shotst->explode.effect1_model, creatng->owner);
+    create_used_effect_or_element(&creatng->mappos, shotst->explode.effect2_model, creatng->owner);
+    if (shotst->explode.around_effect1_model != 0)
+    {
+        create_effect_around_thing(creatng, shotst->explode.around_effect1_model);
+    }
+    if (shotst->explode.around_effect2_model > 0)
+    {
+        create_effect_around_thing(creatng, shotst->explode.around_effect2_model);
+    }
+    kill_creature(creatng, INVALID_THING, -1, CrDed_NoUnconscious);
+    
+    /*
+    if (creature_model_bleeds(creatng->model))
+    {
+        create_effect_around_thing(creatng, TngEff_Blood5);
+    }
+    //create_effect_around_thing(creatng, TngEff_Explosion6);
+    //struct Thing* shotng = create_shot(&creatng->mappos, ShM_TimeBomb, creatng->owner);
+    //if (!thing_is_invalid(shotng))
+    
+    //struct ShotConfigStats* shotst = get_shot_model_stats(ShM_TimeBomb);
+    //HitTargetFlags hit_targets = hit_type_to_hit_targets(shotst->area_hit_type);
+    //struct MagicStats* pwrdynst = get_power_dynamic_stats(PwrK_TIMEBOMB);
+    struct CreatureControl* cctrl = creature_control_get_from_thing(creatng);
+    cctrl->timebomb_death = true; // Should be done through shot properties
```

Comment:
This bit is still commented out. Yes, here we still need to get the death kind from the shot properties.